### PR TITLE
Don't allow "tags" field to be null

### PR DESCRIPTION
**Pull request by kruai**
## Overview

This PR is being created to address [RHCLOUD-14532](https://issues.redhat.com/browse/RHCLOUD-14532).
It changes the functionality to stop allowing the "tags" field to be set to null/None. We used to allow it to be set to null, and would just replace it with `{}` if it was. However, this was planned to be phased out, and the time has come.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Checklist GitHub Link

- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [x] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by kruai** [comment]
I want to add at least another test or two to validate this new restriction, but the functionality coding should be done. Let me know if y'all think the exceptions are redundant at all.

File: app/models.py [hunk 1]
```diff
@@ -304,8 +307,10 @@ def replace_facts_in_namespace(self, namespace, facts_dict):
         orm.attributes.flag_modified(self, "facts")
 
     def _update_tags(self, tags_dict):
-        if not self.tags:  # fixme: Host tags should never be None, in DB neither NULL nor 'null'
-            self.tags = {}
+        if self.tags is None:
+            raise InventoryException(
+                title="Invalid request", detail="Both stale_timestamp and reporter fields must be present."
```
Comment (fstavela) [inline app/models.py:312]: You probably forgot to update the error message here

Comment (kruai) [reply app/models.py:312]: Whoops, I did

Comment (kruai) [reply app/models.py]: Updated!

**Review by ryandillinfelton — APPROVED** [review]
Looks good to me :+1:

**Review by thearifismail — APPROVED** [review]

**Pull request merged**